x-pack/plugins/ml/public/components/anomalies_table/anomalies_table.js

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+/*
+ * React table for displaying a list of anomalies.
+ */
+
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import $ from 'jquery';
+
+import React, {
+  Component
+} from 'react';
+
+import {
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHealth,
+  EuiInMemoryTable,
+  EuiText
+} from '@elastic/eui';
+
+import { formatDate } from '@elastic/eui/lib/services/format';
+
+import { DescriptionCell } from './description_cell';
+import { EntityCell } from './entity_cell';
+import { InfluencersCell } from './influencers_cell';
+import { AnomalyDetails } from './anomaly_details';
+import { LinksMenu } from './links_menu';
+
+import { mlAnomaliesTableService } from './anomalies_table_service';
+import { mlFieldFormatService } from 'plugins/ml/services/field_format_service';
+import { getSeverityColor } from 'plugins/ml/../common/util/anomaly_utils';
+import { formatValue } from 'plugins/ml/formatters/format_value';
+
+
+const INFLUENCERS_LIMIT = 5;    // Maximum number of influencers to display before a 'show more' link is added.
+
+
+function renderTime(date, aggregationInterval) {
+  if (aggregationInterval === 'hour') {
+    return formatDate(date, 'MMMM Do YYYY, HH:mm');
+  } else if (aggregationInterval === 'second') {
+    return formatDate(date, 'MMMM Do YYYY, HH:mm:ss');
+  } else {
+    return formatDate(date, 'MMMM Do YYYY');
+  }
+}
+
+function showLinksMenuForItem(item) {
+  return item.isTimeSeriesViewDetector ||
+    item.entityName === 'mlcategory' ||
+    item.customUrls !== undefined;
+}
+
+function getColumns(
+  items,
+  examplesByJobId,
+  isAggregatedData,
+  interval,
+  timefilter,
+  showViewSeriesLink,
+  itemIdToExpandedRowMap,
+  toggleRow,
+  filter) {
+  const columns = [
+    {
+      name: '',
+      render: (item) => (
+        <EuiButtonIcon
+          onClick={() => toggleRow(item)}
+          iconType={itemIdToExpandedRowMap[item.rowId] ? 'arrowDown' : 'arrowRight'}
+          aria-label={itemIdToExpandedRowMap[item.rowId] ? 'Hide details' : 'Show details'}
+          data-row-id={item.rowId}
+        />
+      )
+    },
+    {
+      field: 'time',
+      name: 'time',
+      dataType: 'date',
+      render: (date) => renderTime(date, interval),
+      sortable: true
+    },
+    {
+      field: 'severity',
+      name: `${(isAggregatedData === true) ? 'max ' : ''}severity`,
+      render: (score) => {
+        return (
```

Comment:
these render functions are a bit inconsistent, the first omits the `return` but the others don't.
Personally I don't thing the `return` is necessary if it's just returning a single component, as in the first column.